pkg/reconciler/broker/broker.go

```diff
@@ -0,0 +1,505 @@
+				err = r.triggerReconciler.Reconcile(ctx, tKey)
+			}
+		}
+	}
+
+	//TODO aggregate errors?
+	return err
+}
+
+//TODO all this stuff should be in a configmap variant of the config object
+
+// This function is not thread-safe and should only be executed by
+// TargetsConfigUpdater
+func (r *Reconciler) updateTargetsConfig(ctx context.Context) error {
+	//Load the existing config first if it exists
+	r.loadConfigOnce.Do(func() {
+		if err := r.loadTargetsConfig(ctx); err != nil {
+			r.Logger.Error("error loading targets config", zap.Error(err))
+			// For some reason the targets config is corrupt, proceed with an
+			// empty one
+			r.targetsConfig = memory.NewEmptyTargets()
+		}
+	})
+	//TODO resources package?
+	data, err := r.targetsConfig.Bytes()
+	if err != nil {
+		return fmt.Errorf("error serializing targets config: %w", err)
+	}
+	desired := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      targetsCMName,
+			Namespace: system.Namespace(),
+		},
+		BinaryData: map[string][]byte{targetsCMKey: data},
+		// Write out the text version for debugging purposes only
+		Data: map[string]string{"targets.txt": r.targetsConfig.String()},
+	}
+
+	r.Logger.Debug("Current targets config", zap.Any("targetsConfig", r.targetsConfig.String()))
+
+	existing, err := r.configMapLister.ConfigMaps(desired.Namespace).Get(desired.Name)
+	if errors.IsNotFound(err) {
+		r.Logger.Debug("Creating targets ConfigMap", zap.String("namespace", desired.Namespace), zap.String("name", desired.Name))
+		_, err = r.KubeClientSet.CoreV1().ConfigMaps(desired.Namespace).Create(desired)
+		if err != nil {
+			return fmt.Errorf("error creating targets ConfigMap: %w", err)
+		}
+	} else if err != nil {
+		return fmt.Errorf("error getting targets ConfigMap: %w", err)
+	}
+
+	r.Logger.Debug("Compare targets ConfigMap", zap.Any("existing", base64.StdEncoding.EncodeToString(existing.BinaryData[targetsCMKey])), zap.String("desired", base64.StdEncoding.EncodeToString(desired.BinaryData[targetsCMKey])))
+	if !equality.Semantic.DeepEqual(desired.BinaryData, existing.BinaryData) {
+		r.Logger.Debug("Updating targets ConfigMap")
+		_, err = r.KubeClientSet.CoreV1().ConfigMaps(desired.Namespace).Update(desired)
+		if err != nil {
+			return fmt.Errorf("error updating targets ConfigMap: %w", err)
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) loadTargetsConfig(ctx context.Context) error {
+	r.Logger.Debug("Loading targets config from configmap")
+	//TODO retry with wait.ExponentialBackoff
+	existing, err := r.configMapLister.ConfigMaps(system.Namespace()).Get(targetsCMName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.targetsConfig = memory.NewEmptyTargets()
+			return nil
+		}
+		return fmt.Errorf("error getting targets ConfigMap: %w", err)
+	}
+
+	targets := &config.TargetsConfig{}
+	data := existing.BinaryData[targetsCMKey]
+	if err := proto.Unmarshal(data, targets); err != nil {
+		return err
+	}
```

Comment:
Might as well add the `NewConfigFromBytes` back. Using proto.Unmarshal creates an asymmetric encoding/decoding experience.